### DISCUSSION TITLE
Can you render specific blocks onto a blank page?

### DISCUSSION THREAD

**Discussion by inf3rnus** [pymupdf/PyMuPDF#2006] [Looking for help]
Hey all/Jorj!

I can't see to find in the docs if it's possible to take let's say a block, and to render all of its characters onto a new pdf page?

I should add that what I'm really trying to do is do this with spans... Normally I'd just use the bbox from the spans to make a crop from the original page... However, bboxes are usually slightly off (which I know is an mupdf upstream thing), so what I'm trying to do is basically render out particular spans in isolation from everything else.

Any info would be greatly appreciated!

Thanks,
Aaron

**Comment by JorjMcKie** [comment]
Having electricity issues, so just in short via phone: extract the text via get_text("dict"), then insert_text() on your new page using some or all properties of the extracted text.

Holen Sie sich Outlook für Android<https://aka.ms/AAb9ysg>
________________________________
From: Aaron V ***@***.***>
Sent: Friday, October 28, 2022 8:58:27 PM
To: pymupdf/PyMuPDF ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [pymupdf/PyMuPDF] Can you render specific blocks onto a blank page? (Discussion #2006)


Hey all/Jorj!

I can't see to find in the docs if it's possible to take let's say a block, and to render all of its characters onto a new pdf page?

Any info would be greatly appreciated!

Thanks,
Aaron

**Reply by JorjMcKie** [reply]
Just another thought:
You can always show any portion (rectangle) of a (source) page on any selected portion of another (target) page living in another document object.
This obsoletes rewriting text with all its pesky details (font selection, correct positioning and what not).
This is offered by `page.show_pdf_page()`. Included are options to show the selected clip rotated by any angle.
Depending on your use case, there is the issue that **everything** covered by the selected clip will be shown - not just text.

**Comment by JorjMcKie** [comment]
Sorry for having been so laconic, now my eletricity issues are under better control again.

> However, bboxes are usually slightly off (which I know is an mupdf upstream thing)

What do you mean? Actually this is not the case ...
